GitCommands/CommitTemplateManager.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using GitUIPluginInterfaces;
+
+namespace GitCommands
+{
+    public interface ICommitTemplateManager
+    {
+        /// <summary>
+        /// Loads commit template from the file specified in .git/config 
+        /// under <c>commit.template</c> setting. 
+        /// </summary>
+        /// <exception cref="FileNotFoundException">The specified template file cannot be found.</exception>
+        /// <returns>The commit template, if it is specified; otherwise <see langword="null"/>.</returns>
+        /// <remarks>
+        /// Template file can be set by the following command:
+        /// <c>$ git config --global commit.template ~/.git_commit_msg.txt</c>
+        /// </remarks>
+        string LoadGitCommitTemplate(string workingDir);
+    }
+
+    public sealed class CommitTemplateManager : ICommitTemplateManager
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IGitModule _module;
+
+
+        public CommitTemplateManager(IGitModule module, IFileSystem fileSystem)
+        {
+            _module = module;
+            _fileSystem = fileSystem;
+        }
+
+        public CommitTemplateManager(IGitModule module)
+            : this(module, new FileSystem())
+        {
+        }
+
+
+        /// <summary>
+        /// Loads commit template from the file specified in .git/config 
+        /// under <c>commit.template</c> setting. 
+        /// </summary>
+        /// <exception cref="FileNotFoundException">The specified template file cannot be found.</exception>
+        /// <returns>The commit template, if it is specified; otherwise <see langword="null"/>.</returns>
+        /// <remarks>
+        /// Template file can be set by the following command:
+        /// <c>$ git config --global commit.template ~/.git_commit_msg.txt</c>
+        /// </remarks>
+        public string LoadGitCommitTemplate(string workingDir)
+        {
+            string fileName = _module.GetEffectiveSetting("commit.template");
+            if (string.IsNullOrEmpty(fileName))
+            {
+                return null;
+            }
+
+            if (!Path.IsPathRooted(fileName))
+            {
+                // git automatically converts '~' if the path contains it
+                // i.e. ~/.template.txt -> C:/Users/<user>/.template.txt (Windows)
+                //                      -> /home/<user>/.gitcommit.txt (Linux)
+                // however if the .gitconfig contains a template starting with '~'
+                // it means the user has manually entered the path
+                // in this case, treat the path as local
+                if (fileName.StartsWith("~"))
```

Comment:
It is almost sure, that there is no such file in the repository. The user will get a FNFE telling him that the file does not exist and giving the full path <repo path>/<fileName>. This will be quite surprising. If you don't want to expand the user home dir (don't know why), then show the original path from the config file.